### ISSUE TITLE
Handling identical file names

### ISSUE THREAD

**Issue by bowencool** [icloud-photos-downloader/icloud_photos_downloader#918]
## Overview

An edge case when using `auto-delete`.

## Steps to Reproduce

1. Edit a video file (let's say it's called 1234.mov) and save it to a new clip, which will also be accurately named 1234.mov on iCloud. (Importing a file twice is also applicable.)
2. Delete the old one on iCloud.
3. run icloudpd with `--auto-delete`.

## Expected Behavior

Keep it exactly the same as the iCloud folder.

## Actual Behavior

iCloudPD will download the new 1234.mov and then delete it because an old file with the same name, 1234.mov, is in iCloud's "Recently Deleted" folder.

## Context
[NOTE]: # ( Give us any additional information you may have. )

**Comment by AndreyNikiforov** [comment]
Yes, assets with the same name and deletions are not working perfectly in deduplication mode (default). [`--file-match-policy name-id7`](https://icloud-photos-downloader.github.io/icloud_photos_downloader/naming.html#duplicates) shuld not have that problem by design, but the automatic migration from one file-match-policy to another is not available yet
